src/instana/instrumentation/couchbase_inst.py

```diff
@@ -6,90 +6,148 @@
 https://docs.couchbase.com/python-sdk/2.5/start-using-sdk.html
 """
 
+from typing import Any, Callable, Dict, Tuple
+
 import wrapt
 
-from ..log import logger
-from ..util.traceutils import get_tracer_tuple, tracing_is_off
+from instana.log import logger
+from instana.span.span import InstanaSpan
+from instana.util.traceutils import get_tracer_tuple, tracing_is_off
 
 try:
     import couchbase
-
-    if not (hasattr(couchbase, '__version__') and couchbase.__version__[0] == '2'
-            and (couchbase.__version__[2] > '3'
-                 or (couchbase.__version__[2] == '3' and couchbase.__version__[4] >= '4'))
-        ):
+    from couchbase.bucket import Bucket
+
+    if not (
+        hasattr(couchbase, "__version__")
+        and couchbase.__version__[0] == "2"
+        and (
+            couchbase.__version__[2] > "3"
+            or (couchbase.__version__[2] == "3" and couchbase.__version__[4] >= "4")
+        )
+    ):
         logger.debug("Instana supports 2.3.4 <= couchbase_versions < 3.0.0. Skipping.")
         raise ImportError
 
     from couchbase.n1ql import N1QLQuery
 
     # List of operations to instrument
     # incr, incr_multi, decr, decr_multi, retrieve_in are wrappers around operations above
-    operations = ['upsert', 'insert', 'replace', 'append', 'prepend', 'get', 'rget',
-                  'touch', 'lock', 'unlock', 'remove', 'counter', 'mutate_in', 'lookup_in',
-                  'stats', 'ping', 'diagnostics', 'observe',
-
-                  'upsert_multi', 'insert_multi', 'replace_multi', 'append_multi',
-                  'prepend_multi', 'get_multi', 'touch_multi', 'lock_multi', 'unlock_multi',
-                  'observe_multi', 'endure_multi', 'remove_multi', 'counter_multi']
-
-    def capture_kvs(scope, instance, query_arg, op):
+    operations = [
+        "upsert",
+        "insert",
+        "replace",
+        "append",
+        "prepend",
+        "get",
+        "rget",
+        "touch",
+        "lock",
+        "unlock",
+        "remove",
+        "counter",
+        "mutate_in",
+        "lookup_in",
+        "stats",
+        "ping",
+        "diagnostics",
+        "observe",
+        "upsert_multi",
+        "insert_multi",
+        "replace_multi",
+        "append_multi",
+        "prepend_multi",
+        "get_multi",
+        "touch_multi",
+        "lock_multi",
+        "unlock_multi",
+        "observe_multi",
+        "endure_multi",
+        "remove_multi",
+        "counter_multi",
+    ]
+
+    def collect_attributes(
+        span: InstanaSpan,
+        instance: Bucket,
+        query_arg: Any,
```

Comment:
Is there something better than `Any`?

Reply:
If there is no way to check which other type is coming, then I prefer `Union[N1QLQuery, object]`.